File: workspace/apps/pidp/src/app/features/access/pages/digital-evidence/case-management/digital-evidence-case.model.ts
export enum CaseStatus {
  NewRequest = 'In Progress',
  RemoveRequested = 'Remove Requested',
  Active = 'Active',
  Pending = 'Pending',
  Completed = 'Complete',
}

export interface Field {
  id: number;
  name: string;
  value: any;
  display: boolean;
}

export interface DigitalEvidenceCase {
  name: string;
  description: string;
  key: string;
  justinStatus: JustinCaseStatus;
  agencyFileNumber: string;
  details: string;
  status: string;
  fields: Field[] | [];
  id: number;
}

export interface JustinCaseStatus {
  value: string;
  demsCandidate: boolean;
  description: string;
}

export interface DigitalEvidenceCaseAccessRequest {
  partyId: number;
  caseId: number;
  key: string;
  toolsCaseRequest: boolean;
  agencyFileNumber: string;
  name: string;
}

export interface DigitalEvidenceCaseRequest extends DigitalEvidenceCase {
  requestedOn: Date;
  requestId: number;
  assignedOn: Date | null;
  removalRequest: boolean;
  requestStatus: CaseStatus;
}

File: backend/webapi/Models/DigitalEvidenceCaseModel.cs
namespace Pidp.Models;

using NodaTime;

/// <summary>
/// Represents a case from another system (e.g. EDT DEMS)
/// </summary>
public class DigitalEvidenceCaseModel
{
    public Common.Models.JUSTIN.CaseStatus? JustinStatus { get; set; }

    public int Id { get; set; }
    public int RequestId { get; set; }
    public int PartyId { get; set; }
    public string Name { get; set; } = string.Empty;

    public string Details { get; set; } = string.Empty;

    public string AgencyFileNumber { get; set; } = string.Empty;
    public string Key { get; set; } = string.Empty;

    public Instant RequestedOn { get; set; }
    public Instant AssignedOn { get; set; }
    public Instant LastUpdated { get; set; }
    public string Status { get; set; } = string.Empty;
    public string Errors { get; set; } = string.Empty;
    public string RequestStatus { get; set; } = string.Empty;

    public List<Field> Fields { get; set; }

}
public class Field

{
    public int Id { get; set; }
    public string Name { get; set; }
    public object Value { get; set; }

    public bool Display { get; set; }
}

public enum DEMSCaseStatus
{
    Active,
    Inactive,
    NotFound,
    Error
}
